src/DotNet.Testcontainers/Clients/TestcontainersClient.cs

```diff
@@ -167,13 +172,22 @@ await this.containers.ExtractArchiveToContainerAsync(id, "/", memStream, ct)
 
     public async Task<string> RunAsync(ITestcontainersConfiguration configuration, CancellationToken ct = default)
     {
+      // Killing or canceling the test process will prevent the cleanup.
+      // Remove labeled, orphaned containers from previous runs.
+      var removeOrphanedContainersTasks = (await this.containers.GetOrphanedObjects(ct)
+        .ConfigureAwait(false))
+        .Select(container => this.containers.RemoveAsync(container.ID, ct));
+
       if (!await this.images.ExistsWithNameAsync(configuration.Image.FullName, ct)
         .ConfigureAwait(false))
       {
         await this.images.CreateAsync(configuration.Image, configuration.AuthConfig, ct)
           .ConfigureAwait(false);
       }
 
+      await Task.WhenAll(removeOrphanedContainersTasks)
```

Comment:
Shouldn't this one be done before the image creation?

Reply:
No. We can pull not existing images while we're removing orphaned containers.